spec/unit/workload_mapper_spec.rb

```diff
@@ -0,0 +1,144 @@
+      {
+        "services": {
+          "init_system": "systemd",
+          "services": [
+            {
+              "name": "mysql.service",
+              "state": "enabled"
+            }]
+        },
+        "config_files": {
+          "extracted": true,
+          "files": [
+            {
+              "name": "/etc/my.cnf"
+            }
+          ]
+        }
+      }
+    EOF
+  }
+  subject { WorkloadMapper.new(system_description) }
+
+  let(:output_path) { given_directory }
+  let(:mapper_path) { given_directory_from_data "mapper" }
+  let(:docker_path) { given_directory_from_data "docker" }
+
+  let(:workloads) { YAML::load(File.read(File.join(docker_path, "workloads.yml"))) }
+  let(:look_up_table) { YAML::load(File.read(File.join(mapper_path, "look_up_table.yml"))) }
+  let(:config_file_path) { File.join(mapper_path, "my.cnf") }
+
+  describe "#write_compose_file" do
+    it "writes a docker-compose.yml" do
+      allow(subject).to receive(:workloads).and_return(workloads)
+      subject.write_compose_file(output_path)
+      expect(File.read(File.join(output_path, "docker-compose.yml"))).
+        to include(File.read(File.join(docker_path, "docker-compose.yml")))
+    end
+  end
+
+  describe "#compose_service" do
+    let(:expected_node) {
+      {
+        "db" => {
+          "image" => "opensuse/mariadb",
+          "environment" => {
+            "MYSQL_USER" => "portus",
+            "MYSQL_PASS" => "portus"
+          }
+        }
+      }
+    }
+    let(:workload) { "mariadb" }
+    let(:config) {
+      {
+        "service" => "db",
+        "parameters" => {
+          "user" => "portus",
+          "password" => "portus"
+        }
+      }
+    }
+
+    it "returns a valid compose node" do
+      allow(subject).to receive(:workloads).and_return(workloads)
+      allow(subject).to receive(:look_up_table).and_return(look_up_table)
+
+      expect(subject.compose_service(workload, config)).to eq(expected_node)
+    end
+  end
+
+  describe "#identify" do
+    context "when no service is specified" do
+      specify "name and service equaliy" do
+        subject.instance_eval do
+          identify "foo"
+        end
+        expect(subject.workloads["foo"]).to include("service" => "foo")
+      end
+    end
+    context "when service is specified" do
+      it "sets the name and the service" do
+        subject.instance_eval do
+          identify "foo", "bar"
+        end
+        expect(subject.workloads["foo"]).to include("service" => "bar")
+      end
+    end
+  end
+
+  describe "#parameter" do
+    it "appends a parameter to the current workflow" do
+      allow(subject).to receive(:workloads).and_return(workloads)
+      allow(subject).to receive(:current_workload).and_return("mariadb")
+
+      subject.instance_eval do
+        parameter "port", "1234"
+      end
+
+      expect(subject.workloads["mariadb"]["parameters"]).to include("port" => "1234")
+    end
+  end
```

Comment:
It would be nice, if the tests for the DSL methods would be less dependent on the implementation. I would imagine the test to more look like the following sketch:

```ruby
  mapper = WorkloadMapper.new

  clue = 'identify "foo"'
  workload = mapper.check_clue(clue)

  expect(workload.name).to eq("foo")
```